src/Gui/PythonWorkbenchPyImp.cpp

```diff
@@ -62,15 +62,37 @@ PyObject*  PythonWorkbenchPy::appendMenu(PyObject *args)
             int nDepth = PyList_Size(pPath);
             for (int j=0; j<nDepth;++j) {
                 PyObject* item = PyList_GetItem(pPath, j);
-                if (!PyString_Check(item))
+                if (PyUnicode_Check(item)) {
+#if PY_MAJOR_VERSION >= 3
+                    char* pItem = PyUnicode_AsUTF8(item);
+                    path.push_back(pItem);
+#else
+                    PyObject* unicode = PyUnicode_AsEncodedString(item, "utf-8", 0);
+                    char* pItem = PyString_AsString(unicode);

```

Comment:
We have to be careful here because when decrementing "unicode" the pItem becomes a dangling pointer. So, you must first push back the pItem to path before destroying unicode. This is safe because the string is then copied and hold in a std::string.